### PR TITLE
Features/slurm

### PR DESCRIPTION
**Pull request by dpopleton**
**Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion**

## Requirements

- Filling out the template is required. 
- Review the [contributing guidelines](https://github.com/Ensembl/ensembl/blob/master/CONTRIBUTING.md#why-could-my-pull-request-be-rejected) for this repository; remember in particular:
    - do not modify code without testing for regression
    - provide simple unit tests to test the changes
    - if you change the schema you must patch the test databases as well, see [Updating the schema](https://github.com/Ensembl/ensembl/blob/master/CONTRIBUTING.md#updating-the-schema)
    - the PR must not fail unit testing

## Description

Allows the implementation of SLURM in beekeeper. The base configuration and the core stats are changed.

## Use case

Will be used for full implementation of SLURM

## Benefits

We will be able to use SLURM

## Possible Drawbacks

More complex code. Could break everything. It has been tested and no issues were discovered.

## Testing

- [ ] Have you added/modified unit tests to test the changes?
- [ ] If so, do the tests pass?
- [ ] Have you run the entire test suite and no regression was detected?
- [ ] TravisCI passed on your branch

Dependencies
------------

> If applicable, define what code dependencies were added and/or updated.

### CODE REVIEW

File: modules/Bio/EnsEMBL/Production/Pipeline/PipeConfig/Base_conf.pm [hunk 1]
```diff
@@ -66,19 +66,57 @@ sub beekeeper_extra_cmdline_options {
...
+        $output{$memory_key} = {'LSF' => '-q '.$self->o('production_queue').' -M '.$memory_value.' -R "rusage[mem='.$memory_value.']"',
+                                'SLURM' => $pq.$time_value.'  --mem='.$memory_value.'m' }
+      }else{
+        $output{$memory_key.'_'.$time_key} = {'LSF' => '-q '.$self->o('production_queue').' -M '.$memory_value.' -R "rusage[mem='.$memory_value.']"',
+                                'SLURM' => $pq.$time_value.'  --mem='.$memory_value.'m' }
+      }
+    }
   }
+
+return %output;
```
Comment (EbiArnie) [inline modules/Bio/EnsEMBL/Production/Pipeline/PipeConfig/Base_conf.pm]: This should return a hashref, I think.
return \%output;

**Review by EbiArnie — APPROVED** [review]
OK

File: modules/Bio/EnsEMBL/Production/Pipeline/PipeConfig/Base_conf.pm [hunk 2]
```diff
@@ -30,55 +29,94 @@ use Bio::EnsEMBL::Hive::Version 2.5;
...
+        '8GB'            => '8000',
+        '16GB'           => '16000',
+        '32GB'           => '32000',);
+
+    my $pq = ' --partition=standard';
+    my $dq = ' --partition=datamover';
+
+    my %output = (
+        #Default is a duplicate of 100M
+        'default'   => { 'LSF' => '-q ' . $self->o('production_queue'), 'SLURM' => $pq . $time{'H'} . ' --mem=' . $memory{'100M'} . 'm' },
```
[the remark this answers is not in the document]
Comment (dpopleton) [reply modules/Bio/EnsEMBL/Production/Pipeline/PipeConfig/Base_conf.pm:91]: Most actually would fit in a minute. However hour/minute don't make a difference on current set up. So yes

[the remark this answers is not in the document]
Comment (marcoooo) [reply modules/Bio/EnsEMBL/Production/Pipeline/PipeConfig/Base_conf.pm:91]: ok, all fine then. One more question regarding the other defaults we had before, for code portability, would it be possible to default the others as well? I mean to keep the previous entries but with default time?

[the remark this answers is not in the document]
Comment (dpopleton) [reply modules/Bio/EnsEMBL/Production/Pipeline/PipeConfig/Base_conf.pm:91]: Yes. They are all there. Check out line 108 of base_conf. Right now our values are default, and then variants on memory (8GB) etc. So they should all be fine if they run in an hour.
One thing I did that is perhaps unnecessary is to add the specific resource class to every pipeline analysis. This was just to keep track, but I prefer it this way. Ex:      -rc_name         => 'default',

**Pull request closed without merging**